Subject: display last accessed sysfs file on kernel panic message
From: Andrew Morton <akpm@osdl.org>
Patch-mainline: never

Display the most-recently-opened sysfs file's name when oopsing.

From: Adrian Bunk <bunk@stusta.de>

  Build fix

From: Greg Kroah-Hartman <gregkh@suse.de>

  Modified to make the api call cleaner, and available to all arches if
  need be.  Also added it to x86-64's crash dump message.


Signed-off-by: Adrian Bunk <bunk@stusta.de>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
Automatically created from "patches.drivers/sysfs-crash-debugging.patch" by xen-port-patches.py

--- a/arch/i386/kernel/traps-xen.c	2007-07-08 19:32:17.000000000 -0400
+++ b/arch/i386/kernel/traps-xen.c	2007-08-27 14:02:24.000000000 -0400
@@ -411,6 +411,7 @@ void die(const char * str, struct pt_reg
 #endif
 		if (nl)
 			printk("\n");
+		sysfs_printk_last_file();
 		if (notify_die(DIE_OOPS, str, regs, err,
 					current->thread.trap_no, SIGSEGV) !=
 				NOTIFY_STOP) {
--- a/arch/x86_64/kernel/traps-xen.c	2007-07-08 19:32:17.000000000 -0400
+++ b/arch/x86_64/kernel/traps-xen.c	2007-08-27 14:02:23.000000000 -0400
@@ -516,6 +516,7 @@ void __kprobes __die(const char * str, s
 	printk("DEBUG_PAGEALLOC");
 #endif
 	printk("\n");
+	sysfs_printk_last_file();
 	notify_die(DIE_OOPS, str, regs, err, current->thread.trap_no, SIGSEGV);
 	show_registers(regs);
 	/* Executive summary in case the oops scrolled away */
